spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/tf/Session.scala

```diff
@@ -254,27 +255,59 @@ class BigDLSessionImpl[T: ClassTag](graph: Seq[NodeDef], context: Context[T],
     result
   }
 
+  private def allHDFSFiles(fileNames: Seq[String]) = {
+    val isHdfs = fileNames.map(_.startsWith("hdfs:"))
+    if (isHdfs.reduceLeft(_ && _)) {
+      true
+    } else if (isHdfs.map(!_).reduceLeft(_ && _)) {
+      false
+    } else {
+      throw new IllegalArgumentException("filenames contain both local and hdfs path")
+    }
+  }
+
   private def readTFRecord(filesTable: Seq[Table], sc: SparkContext): RDD[Table] = {
-    val result = filesTable.map { t =>
-        require(t.length() == 1 && t(1).isInstanceOf[Tensor[ByteString]],
-          "Reader can only read one file at a time")
-        val fileTensor = t[Tensor[ByteString]](1)
-        require(fileTensor.isScalar)
-        val file = fileTensor.value()
-        file
-    }.flatMap { file =>
-      val iter = new TFRecordIterator(new java.io.File(file.toStringUtf8))
-      iter
-    }.map { record =>
-      val table = T()
-      val key = Tensor[ByteString](Array(ByteString.copyFromUtf8("somekey")), Array[Int]())
-      val value = Tensor[ByteString](Array(ByteString.copyFrom(record)), Array[Int]())
-      table.insert(key)
-      table.insert(value)
-      table
+    if (filesTable.isEmpty) {
+      return sc.parallelize(Seq.empty[Table], numSlices = Engine.coreNumber())
+    }
+    val fileNames = filesTable.map { t =>
+      require(t.length() == 1 && t(1).isInstanceOf[Tensor[ByteString]],
+        "Reader can only read one file at a time")
+      val fileTensor = t[Tensor[ByteString]](1)
+      require(fileTensor.isScalar)
+      val file = fileTensor.value()
+      file.toStringUtf8
+    }
+
+    val isHdfs = allHDFSFiles(fileNames)
+    if (isHdfs) {
+      // all files are hdfs files
+      fileNames.map { fileName =>
+        val rdd = sc.newAPIHadoopFile[BytesWritable, NullWritable, TFRecordInputFormat](fileName)
+        rdd.map { case (k, v) =>
+          val table = T()
+          val key = Tensor.scalar[ByteString](ByteString.copyFromUtf8("fake_key"))
+          val value = Tensor.scalar[ByteString](ByteString.copyFrom(k.copyBytes()))
+          table.insert(key)
+          table.insert(value)
+          table
+        }
+      }.reduceLeft(_.union(_))
+    } else {
+      // all files are local files
+      val result = fileNames.flatMap { file =>
```

Comment:
what if some are hdfs and some are local

Reply:
It will throw an exception at line `val isHdfs = allHDFSFiles(fileNames)`. Is that alright?